Feature: add setting to disable news about competitor's accidents

grenoult opened the pull request:
## Motivation / Problem

During a game of OpenTTD with many competitors, we can receive a lot of news regarding competitors vehicles involved in accidents. This is especially true when vehicles have low reliability and are rarely replaced:
- road vehicles breaking down on a rail crossing then being crashed by a train,
- airplanes failing to land on an airport and crashing.

This PR addresses the issue of receiving too many news regarding competitors vehicle crash where it doesn't bring any value or meaning to the main player. As a OpenTTD player, I want to know when my vehicles are involved in a crash but I may choose to not receive notification if competitors vehicles are destroyed.

## Description

A new setting has been added "Accidents of competitor's vehicles". It's "Full" by default. When turned off, no news will be generated when a competitor vehicle is involved in a crash. When it's "Summary", only the short version will be displayed at the bottom of the screen.
A crash, in the context of this PR, is when a road vehicle is hit by a train or when an airplane misses landing at an airport.

- Full: A news item is generated when setting is Full (by default) and a competitor vehicle is crashed (in this example, a plane crash):
![image](https://user-images.githubusercontent.com/16795616/138574752-65cf20e3-9e73-4e35-a6d3-8d90b1dd63a0.png)

- Summary: Short news are generated when setting is "Summary" and a competitor vehicle (red) is crashed by a train:
![image](https://user-images.githubusercontent.com/16795616/138574704-a6af3633-62aa-4f81-8158-459ad12af3ef.png)

- Off: no news are generated when a competitor vehicle is crashed when settings is "Off"
![image](https://user-images.githubusercontent.com/16795616/138574780-9d3edfad-3ee5-4653-9843-b58f85384d4f.png)


## Limitations

- other crashes, such as UFOs and similar events are ignored as part of this PR. This has been left out because it's rare enough to cause a disruption to the player. That being said, one could argue we could update this in this PR and I could update it (probably in another PR) if requested.

## Checklist for review

- This PR adds a new translation in english.txt file. From the [documentation](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md#i-want-to-add-a-new-string), nothing else is required at this stage.

grenoult committed:
Update: add setting to hide news for competitor road vehicle crash

grenoult committed:
Add setting to hide news for competitor road vehicle crash

grenoult committed:
Do not show competitors airplane crash if disabled in settings

grenoult committed:
Update: do not show competitors airplane crash if disabled in settings

grenoult committed:
Update: remove TODOs

grenoult committed:
Remove TODOs

frosch123 commented:
For "arrival of first vehicle" and "production changes of industries" there are separate settings with "full"/"summary"/"off".
Shouldn't this apply to "accidents" as well?
I guess "disasters" (ufos etc) would be a separate setting as well then.

James103 commented on src/settings_type.h:
```diff
@@ -242,6 +242,7 @@ struct NewsSettings {
 	uint8 arrival_player;                                 ///< NewsDisplay of vehicles arriving at new stations of current player
 	uint8 arrival_other;                                  ///< NewsDisplay of vehicles arriving at new stations of other players
 	uint8 accident;                                       ///< NewsDisplay of accidents that occur
+	bool accident_other_company_vehicle;				  ///< if a vehicle from another company is involved in an accident, show a news item
```
```suggestion
	uint8 accident_other_company_vehicle;                  ///< NewsDisplay of accidents involving competitor's vehicles
```

James103 replied:
This appears to be resolved in your latest commit.

grenoult commented:
> For "arrival of first vehicle" and "production changes of industries" there are separate settings with "full"/"summary"/"off". Shouldn't this apply to "accidents" as well? I guess "disasters" (ufos etc) would be a separate setting as well then.

Good point @frosch123 , I'll update my PR, thanks.

grenoult committed:
Update: switch from a boolean parameter to enum for consistency

To make this setting more consistent with the other settings, moved to a enum type (full, summary, off) and close to Accident / Disasters settings

grenoult committed:
Switch from a boolean parameter to enum for consistency

To make this setting more consistent with the other settings, moved to a enum type (full, summary, off) and close to Accident / Disasters settings

grenoult committed:
Merge branch 'disable_competitor_accident_news_patch' of https://github.com/grenoult/OpenTTD into disable_competitor_accident_news_patch

grenoult commented:
Closing this branch. A new PR has been created with correct commit messages: https://github.com/OpenTTD/OpenTTD/pull/9653

LordAro commented:
(For future reference, you can fix this with [rebasing](https://git-scm.com/docs/git-rebase) without having to open a new PR/make a new branch)
